drivers/platform/stm32/stm32_pwm.c

```diff
@@ -170,6 +172,35 @@ static int32_t stm32_init_timer(struct stm32_pwm_desc *desc,
 	desc->clock_divider = sparam->clock_divider;
 	desc->timer_autoreload = sparam->timer_autoreload;
 
+	if (param->pwm_callback) {
+		ret = lf256fifo_init(&sparam->rx_fifo);
+		if (ret < 0)
+			goto error;
```

Comment:
Why is this FIFO required?

Reply:
This is actually not required. Removed .